src/main/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActor.scala

```diff
@@ -231,9 +231,15 @@ private class AppTaskLauncherActor(
   private[this] def receiveTaskLaunchNotification: Receive = {
     case TaskOpSourceDelegate.TaskOpRejected(op, reason) if inFlight(op) =>
       removeTask(op.taskId)
-      tasksToLaunch += 1
       log.info("Task op '{}' for {} was REJECTED, reason '{}', rescheduling. {}",
         op.getClass.getSimpleName, op.taskId, reason, status)
+
+      op match {
```

Comment:
While testing this I realized that the above line `removeTask(op.taskId)` is also wrong in case the op is a TaskOp.Launch – only the launch was rejected, the Task.Reserved should stay in the tasksMap …